tmk_core/protocol/lufa/adafruit_ble.cpp

```diff
@@ -637,9 +642,7 @@ void adafruit_ble_task(void) {
     if (timer_elapsed(state.last_battery_update) > BatteryUpdateInterval && resp_buf.empty()) {
         state.last_battery_update = timer_read();
 
-        if (at_command_P(PSTR("AT+HWVBAT"), resbuf, sizeof(resbuf))) {
-            state.vbat = atoi(resbuf);
-        }
+        state.vbat = analogRead(BATTERY_LEVEL_PIN);
```

Comment:
Wouldn't the `readPin` GPIO control function work better for this? 
https://docs.qmk.fm/#/internals_gpio_control

Reply:
I remember having tried `readPin`, but my recollection is that it didn't work because it is digital rather than analog, and analog is needed here. (Am I mistaken on this?)

Reply:
Ah, yeah, I think you're right.